client/reducers/users.js

```diff
@@ -1,50 +1,47 @@
 import { GET_USERS, SET_USER_ATS } from '../actions/types';
 
-// TODO: Change this into a object key'd by user id?
 const initialState = {
-    users: []
+    usersById: {}
 };
 
 export default (state = initialState, action) => {
     switch (action.type) {
         case GET_USERS: {
             const users = action.payload;
-            return Object.assign({}, state, {
-                users
+            const usersById = {};
+            users.map(u => {
+                usersById[u.id] = u;
             });
```

Comment:
```js
const usersById = users.reduce((accum, user) => {
  accum[u.id] = u;
  return accum;
}, {});
```

You're not mapping an array's values to a new array, you're reducing an array to object.